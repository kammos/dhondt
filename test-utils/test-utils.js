import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { calculatorReducer } from '../src/calculator';


const customRender = (ui, { initialState, store, ...options } = {}) => {
    const actualStore = store ?? configureStore({
        reducer: { calculator: calculatorReducer },
        preloadedState: initialState,
    });

    const Wrapper = ({ children }) => {
        return <Provider store={actualStore}>
            {children}
        </Provider>
    }

    return render(ui, { wrapper: Wrapper, ...options });
}


export * from '@testing-library/react';
export { customRender as render };