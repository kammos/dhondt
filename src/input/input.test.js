/**
 * @jest-environment jsdom
 */
import { render } from 'test-utils';
import { Input } from '.';

describe('Input component', () => {
    it('should support text inputs', () => {
        const ui = render(<Input name="input" value="value" readOnly/>);

        expect(ui.container).toMatchSnapshot();
    });

    it('should support number inputs', () => {
        const ui = render(<Input name="input" type="number" value="100" readOnly/>);

        expect(ui.container).toMatchSnapshot();
    });

    it('should support checkboxes', () => {
        const ui = render(<Input name="input" type="checkbox" readOnly/>);

        expect(ui.container).toMatchSnapshot();
    });

    it('should remove floating point values', () => {
        const ui = render(<Input name="input" type="number" value="100.1" readOnly/>);

        const input = ui.getByLabelText(/Input/);
        expect(input.value).toBe('');
    });
});