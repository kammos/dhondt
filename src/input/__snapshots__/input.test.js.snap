// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should support checkboxes 1`] = `
<div>
  <label
    class="label checkbox"
  >
    <span
      class="labelText"
    >
      Input
      :
    </span>
    <input
      class="input"
      name="input"
      readonly=""
      type="checkbox"
      value=""
    />
  </label>
</div>
`;

exports[`Input component should support number inputs 1`] = `
<div>
  <label
    class="label number"
  >
    <span
      class="labelText"
    >
      Input
      :
    </span>
    <input
      class="input"
      name="input"
      readonly=""
      type="number"
      value=""
    />
  </label>
</div>
`;

exports[`Input component should support text inputs 1`] = `
<div>
  <label
    class="label undefined"
  >
    <span
      class="labelText"
    >
      Input
      :
    </span>
    <input
      class="input"
      name="input"
      readonly=""
      value="value"
    />
  </label>
</div>
`;
